ee/api/ee_property_definition.py

```diff
@@ -41,7 +43,32 @@ class Meta:
             "verified_by",
         ]
 
-    def update(self, property_definition: EnterprisePropertyDefinition, validated_data):
+    def validate(self, data):
+        validated_data = super().validate(data)
+
+        if "hidden" in validated_data and "verified" in validated_data:
+            if validated_data["hidden"] and validated_data["verified"]:
+                raise serializers.ValidationError("A property cannot be both hidden and verified")
+
+        # If setting hidden=True, ensure verified becomes false
+        if validated_data.get("hidden", False):
+            validated_data["verified"] = False
+        # If setting verified=True, ensure hidden becomes false
+        elif validated_data.get("verified", False):
+            validated_data["hidden"] = False
+
+        return validated_data
+
+    def update(self, property_definition: EnterprisePropertyDefinition, validated_data: dict):
+        # If setting hidden=True, ensure verified becomes false
+        if validated_data.get("hidden", False):
```

Comment:
Why is this logic duplicated here? `validate` should be called before `update` so if it was all in `validate` then should be covered?

Reply:
I don't know 🙈 

@Twixes how are the multiple API endpoints used - there is one for ee and one for non-ee. My ee tests didn't pass without the ee api changes so I just copied them over...